openstack/blockstorage/extensions/backups/results.go

```diff
@@ -158,3 +158,39 @@ func (r commonResult) ExtractInto(v interface{}) error {
 func ExtractBackupsInto(r pagination.Page, v interface{}) error {
 	return r.(BackupPage).Result.ExtractIntoSlicePtr(v, "backups")
 }
+
+// VolumeBackupExt represents a decorated form of a Volume with based on the
+// "backup" extension.
+type VolumeBackupExt struct {
+	// Specifies the backup ID, from which the volume was restored.
+	BackupID string `json:"backup_id"`
```

Comment:
Since this is extending the main Volume result, we can add this directly to `openstack/blockstorage/v3/volumes/results.Volume` as:

```
BackupID *string `json:"backup_id"`
```

More details [here](https://github.com/gophercloud/gophercloud/pull/1854) and [here](https://github.com/gophercloud/gophercloud/blob/master/docs/MICROVERSIONS.md)